File: src/main/java/de/steklopod/sec12/Lec02CtxRateLimiterDemo.java
package de.steklopod.sec12;

import de.steklopod.courseutil.Util;
import de.steklopod.sec12.helper.BookService;
import de.steklopod.sec12.helper.UserService;
import reactor.util.context.Context;

public class Lec02CtxRateLimiterDemo {

    public static void main(String[] args) {

        BookService.getBook()
                .repeat(3)
                .contextWrite(UserService.userCategoryContext())
                .contextWrite(Context.of("user", "mike"))
                .subscribe(Util.subscriber());





    }


}

File: src/main/kotlin/de/steklopod/sec12/Lec02CtxRateLimiterDemo.kt
package de.steklopod.sec12

import de.steklopod.sec12.helper.BookService
import de.steklopod.sec12.helper.UserService
import de.steklopod.utils.Util.subscriber
import reactor.util.context.Context

object Lec02CtxRateLimiterDemo {
    @JvmStatic
    fun main(args: Array<String>) {
        BookService.book()
            .repeat(3)
            .contextWrite(UserService.userCategoryContext())
            .contextWrite(Context.of("user", "mike"))
            .subscribe(subscriber())
    }
}
